packages/calling/src/CallHistory/CallHistory.test.ts

```diff
@@ -115,5 +127,129 @@ describe('Call history tests', () => {
+        Promise.resolve({
+          status: 200,
+          ok: true,
+          json: () => Promise.resolve(MOCK_UPDATE_MISSED_CALL_RESPONSE),
+        })
+      ) as jest.Mock;
+    });
+
+    it('successfully updates missed calls', async () => {
+      const endTimeSessionIds = [{endTime: '1234568', sessionId: '123'}];
+      const response = await callHistory.updateMissedCalls(endTimeSessionIds);
+      const convertedEndTimeSessionIds = endTimeSessionIds.map((session) => ({
+        ...session,
+        endTime: new Date(session.endTime).getTime(),
+      }));
+      expect(response.statusCode).toEqual(200);
+      expect(response).toEqual(MOCK_UPDATE_MISSED_CALL_RESPONSE);
+      expect(global.fetch).toHaveBeenCalledTimes(1);
+      expect(global.fetch).toHaveBeenCalledWith(janusSetReadStateUrl, {
+        method: HTTP_METHODS.POST,
+        headers: {
+          [CONTENT_TYPE]: APPLICATION_JSON,
+          Authorization: await webex.credentials.getUserToken(),
+        },
+        body: JSON.stringify({endTimeSessionIds: convertedEndTimeSessionIds}),
+      });
+    });
+
+    it('Error: updateMissedCalls throw 400 error', async () => {
+      const endTimeSessionIds = [];
+      global.fetch = jest.fn(() =>
+        Promise.resolve({
+          status: 400,
+          ok: false,
+        })
+      ) as jest.Mock;
+      const response = await callHistory.updateMissedCalls(endTimeSessionIds);
+      const convertedEndTimeSessionIds = endTimeSessionIds.map((session) => ({
+        ...session,
+        endTime: new Date(session.endTime).getTime(),
+      }));
+      expect(response).toStrictEqual(ERROR_DETAILS_400);
+      expect(response.statusCode).toBe(400);
+      expect(global.fetch).toHaveBeenCalledTimes(1);
+      expect(global.fetch).toHaveBeenCalledWith(janusSetReadStateUrl, {
+        method: HTTP_METHODS.POST,
+        headers: {
+          [CONTENT_TYPE]: APPLICATION_JSON,
+          Authorization: await webex.credentials.getUserToken(),
+        },
+        body: JSON.stringify({endTimeSessionIds: convertedEndTimeSessionIds}),
+      });
+      expect(serviceErrorCodeHandlerSpy).toBeCalledOnceWith(
+        {
+          statusCode: 400,
+        },
+        {
+          file: CALL_HISTORY_FILE,
+          method: 'updateMissedCalls',
+        }
+      );
+    });
+
+    it('Error: updateMissedCalls throw 401 error', async () => {
+      const endTimeSessionIds = [];
+      global.fetch = jest.fn(() =>
+        Promise.resolve({
+          status: 401,
+          ok: false,
+        })
+      ) as jest.Mock;
+
+      const response = await callHistory.updateMissedCalls(endTimeSessionIds);
+      const convertedEndTimeSessionIds = endTimeSessionIds.map((session) => ({
+        ...session,
+        endTime: new Date(session.endTime).getTime(),
+      }));
+      expect(response).toStrictEqual(ERROR_DETAILS_401);
+      expect(response.statusCode).toBe(401);
+      expect(global.fetch).toHaveBeenCalledTimes(1);
```

Comment:
same comment here as above

Reply:
Done

Reply:
This is not addressed

Reply:
Also since the second argument here is the same object every time, instead of having all the fields to be copied again and again. Create an object to store it's value and pass that here

Reply:
Done